src/tests/UnitTests/EnsureEnumParamTests.cs

```diff
@@ -9,29 +9,121 @@ public class EnsureEnumParamTests : UnitTestBase
         [Fact]
         public void IsDefined_ShouldNotThrow()
         {
-            var item = Only1IsValid.Valid;
+            var item = EnumDummy.Valid;
 
             ShouldNotThrow(
                 () => Ensure.Enum.IsDefined(item, ParamName),
                 () => EnsureArg.EnumIsDefined(item, ParamName),
                 () => Ensure.That(item, ParamName).IsDefined());
         }
 
+        [Theory]
+        [InlineData((EnumDummy)2)]
+        [InlineData((EnumDummy)0)]
+        public void NotDefined_ShouldThrow(object itemAsObject)
```

Comment:
Goes for similar tests. Why the need for `object` and then directly casting to the correct type. I've tried with defining the `EnumDummy` as argument and it works fine.

Reply:
If you're OK with exposing that class as public, then it could be used directly. Otherwise:
![image](https://user-images.githubusercontent.com/17761878/75220035-2cbd9b80-57a7-11ea-85f7-31b65f8e9c5c.png)

Reply:
I'd expose it as public, doesn't hurt anything from the test project. Interesting gotcha on the data driven tests though!